projects/plugins/jetpack/modules/masterbar/admin-menu/class-wpcom-admin-menu.php

```diff
@@ -246,18 +248,49 @@ public function add_stats_menu() {
 		add_menu_page( __( 'Stats', 'jetpack' ), $menu_title, 'edit_posts', 'https://wordpress.com/stats/day/' . $this->domain, null, 'dashicons-chart-bar', 3 );
 	}
 
+	/**
+	 * Holds the current product, set by get_current_product().
+	 *
+	 * @var string
+	 */
+	private $cached_product = null;
+	/**
+	 * Gets the current product and stores it in $cached_product so the database is only called once per request.
+	 *
+	 * @return array
+	 */
+	private function get_current_product() {
+		if ( null === $this->cached_product && class_exists( 'WPCOM_Store_API' ) ) {
+			$this->cached_product = \WPCOM_Store_API::get_current_plan( get_current_blog_id() );
```

Comment:
```suggestion
			$this->cached_product[get_current_blog_id()] = \WPCOM_Store_API::get_current_plan( get_current_blog_id() );
```
Can we use an array so it cache busts on blog switch?

Reply:
Since the variable only holds the value for the current request, I don't think we'll have a problem. But, probably a good idea just in case. ✅